lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -244,8 +226,119 @@ def _run(self, params, concierge_params=None):
+    #     ]
+    #     self.logger.debug(
+    #         f"Took {time.time() - workspace_check_time} to check all workspace objects"
+    #     )
+    #     #Get a mapping of condor resources
+    #     catalog_check_time = time.time()
+    #     condor_resources_mapping = self.catalog_utils.get_mass_resources(
+    #         job_param_set=job_param_set, condor=self.condor
+    #     )
+    #     self.logger.debug(
+    #         f"Took {time.time() - catalog_check_time} to get all condor resources"
+    #     )
+    #     # Get a mapping of method git versions. If one is not valid, throw the catalog error #TODO Test this
+    #     catalog_check_time_git = time.time()
+    #
+    #
+    #     return JobResourceMappings(
+    #         condor_resources=condor_resources_mapping,
+    #         method_service_versions=service_versions_mapping,
+    #     )
+
+    def _evaluate_job_params_set(self, job_param_set, parent_job):
+        # Fail early before job submission if possible
+        # Make sure clientgroups and condor resources are set correctly for each job,
+        # or raise an exception if something goes wrong
+        for job_params in job_param_set:
+            self.catalog_utils.get_git_commit_version(job_params=job_params)
+            self.catalog_utils.get_condor_resources(
+                job_params=job_params, condor=self.condor
+            )
+            # Don't allow differing parent jobs, and ensure job always runs with the parent
+            if (
+                "parent_job_id" in job_params
+                and job_params["parent_job_id"] != parent_job.id
+            ):
+                raise MultipleParentJobsException(
+                    "Launching child jobs with differing parents is not allowed"
+                )
+
+    def _run_batch(self, parent_job: Job, job_param_set: List[Dict]) -> List:
+        # Prepare jobs, fail early if possible
+        self._evaluate_job_params_set(job_param_set, parent_job)
+
+        child_job_ids = []
+        # Initiatialize Job Records
+        for job_params in job_param_set:
+            job_params["parent_job_id"]: parent_job.id
+            resources = self.catalog_utils.get_condor_resources(
+                job_params=job_params, condor=self.condor
+            )
+            try:
+                child_job_id = self._init_job_rec(
+                    user_id=self.sdkmr.user_id,
+                    params=job_params,
+                    resources=resources,
+                )
+                child_job_ids.append(child_job_id)
+            except Exception as e:
+                self.logger.debug(
+                    msg=f"Failed to submit child job. Aborting entire batch job {e}"
+                )
+                self._abort_child_jobs(child_job_ids)
+                raise e
+        # Save record of child jobs
+
+        parent_job.child_jobs = child_job_ids
+        parent_job.save()
+
+        # TODO Launch Job Submission Thread
+        return child_job_ids
```

Comment:
I don't think I quite understand what's going on here well enough to review it properly. here's the differences I see in the method other than _evaluate...:

1) `job_params['parent_job_id']` is now always overwritten by `parent_job.id`, regardless of whether the key is in the dict or not
2) condor resources are fetched for each child job, presumably because of 3)
3) rather than `_run`ing the child job, the job is `_init_job_rec`ed. It sounds like the job is no longer run (?) and I don't see any code changes elsewhere that would run it, which implies to me that the job is not longer run, which seems incorrect. Or maybe that's what the TODO is about. If so, shouldn't tests be failing?

It'd be super helpful to get a quick explanation of what's going on

Reply:
1) correct. the child jobs need to know what their parent is, and we set the parent
2) they are fetched from the catalog and cached
3) correct, as mentioned during standup at some point, this seemed like a good break-off point to prevent a massive PR. The tests wouldn't fail, since they test for ee2 job ids. I don't see a way for the tests to know if the job was ever submitted to condor in a different thread unless we ran some timer and looped over check_job, and failed if the status was never updated.

Reply:
> The tests wouldn't fail, since they test for ee2 job ids.

Reply:
> The tests wouldn't fail, since they test for ee2 job ids.

Meaning the tests only check that there's a record in the database for the job, and not that it's in any particular state? E.g. the tests would pass if the job was `_init_job_rec`d, `_run`d, completed, errored, or whatever?

Reply:
I'm still trying to understand the overwrite. Previously there was a check such that it wasn't overwritten if the key was already there. What was the purpose of that check, and why is it no longer needed?

Reply:
the reason for removing the check on overwriting parent id is because we're no longer supporting a batch with more than one parent job.

Reply:
Tests just check for job IDs. State transitions are not expected, and condor is ignored in the test so in that context there's no difference between run job and init job.